packages/components/src/utils/OSfunctions.tsx

```diff
@@ -110,3 +110,29 @@ export function useExternalLink(): (
     }
   }
 }
+
+/**
+ * Returns whether or not the device has a home button
+ */
+export const hasHomeButton = () => {
+  const { height, width } = Dimensions.get('window')
+
+  if (Platform.OS !== 'ios') {
+    // Only iOS devices have this concern
+    return false
+  }
+
+  const devicesWithHomeButton = [
+    { height: 667, width: 375 }, // iPhone 6, 6s, 7, 8, SE (2nd gen)
+    { height: 736, width: 414 }, // iPhone 6+, 6s+, 7+, 8+
+    { height: 568, width: 320 }, // iPhone 5, 5s, 5c, SE (1st gen)
+  ]
+
+  const currentDevice = { height, width }
+
+  return devicesWithHomeButton.some(
+    (device) =>
+      device.height === currentDevice.height &&
+      device.width === currentDevice.width,
+  )
```

Comment:
I'm not sure about this. It feels brittle like new devices (e.g. iPhone SE 4th gen) could have a new resolution and things could break on us--especially since the list doesn't note the SE (3rd gen), but looks like it has the same resolution as the 2nd gen.

Generally: not sure how I feel about us being in the business of manually handling this.

Reply:
Do you have an alternative suggestion? iPhone SE 4th gen [is rumored](https://www.macrumors.com/guide/iphone-se-4/) to no longer have a home button so this list shouldn't change.

Reply:
We couldn't do native module stuff at all without converting the project, right? Had found [this](https://stackoverflow.com/questions/52652956/how-to-detect-if-the-device-iphone-has-physical-home-button) on basically querying the insets without directly using them.

Otherwise I guess not if not leaning on `react-native-safe-area-context` in some capacity, whether directly or using as a means to do a similar query to the above without native.